python/tests/stress/arcticdb/test_mem_leaks.py

```diff
@@ -0,0 +1,201 @@
+"""
+Copyright 2024 Man Group Operations Limited
+
+Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+
+As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+"""
+
+import time
+import psutil
+import gc
+import os
+import numpy as np
+import pandas as pd
+import pytest
+import arcticdb as adb
+from pympler import muppy, summary
+
+from arcticdb.util.test import get_sample_dataframe
+from arcticdb.version_store._store import NativeVersionStore
+
+
+def get_process_by_id(id: int | None) -> psutil.Process:
+    """
+        Returns the id of a process. Call with 'os.getpid()' for current python process
+    """
+    for proc in psutil.process_iter(['pid', 'name', 'memory_info']):
+        if id == proc.pid :
+            return proc
+    return None
+
+def nice_bytes_str(bytes) :
+   return f" {bytes / (1024 * 1024):.2f}MB/[{bytes}] "
+
+def lets_collect_some_garbage(time_sec:int = 7):
+    """
+       Do a garbage collection
+    """
+    gc.collect()
+    print(f"Lets pause for {time_sec} secs so GC to kick in") 
+    time.sleep(time_sec)
```

Comment:
Why do you need a sleep?

Reply:
yes. gc.collect() seems to be asynchronous - in other words the more you wait the better is the chance to go as low as possible. That is the reason I needed to introduce waiting. Without it sometimes almost nothing is collected. And yes it will add some latency to the test for sure ...

Memray tells similar story

https://pytest-memray.readthedocs.io/en/latest/usage.html
.....
Warning

As the Python interpreter has its own [object allocator](https://docs.python.org/3/c-api/memory.html) it’s possible that memory is not immediately released to the system when objects are deleted, so tests using this marker may need to give some room to account for this.